### PR TITLE
Jwt token

### PR DESCRIPTION
**Pull request by goopnigoop**
[EPMLABSBRN-353](https://jira.epam.com/jira/browse/EPMLABSBRN-353)

**Description**: Authentication moved to jwt instead of basic auth.
**--Interim solution, potential migration to auth. server--**
What is done:

- [x] Added jwt support
- [x] autogeneration of jwt based on JwtConfig values
- [x] added JwtUsernameAndPasswordAuthenticationFilter to generate jwt token after successful username and password authentication
- [x] added JwtTokenAuthenticationFilter for jwt verification 
- [x] added kotlinObjectMapper as an extension of objectMapper to avoid deserialization issues with kotlin data classes
- [x] necessary changes in the frontend

TBD:
- [ ] for admin module, changed mocked token to  JWT -> needed to know whether it's still used
- [ ] add logic for expiration part

### CODE REVIEW

File: src/test/kotlin/com/epam/brn/integration/AuthorizationAuthenticationIT.kt [hunk 1]
```diff
@@ -96,16 +94,6 @@ class AuthorizationAuthenticationIT {
         resultAction.andExpect(status().`is`(403))
     }
 
-    @Test
```
Comment (ElenaSpb) [inline src/test/kotlin/com/epam/brn/integration/AuthorizationAuthenticationIT.kt:99]: why have you delete such useful test?

File: src/test/kotlin/com/epam/brn/integration/AuthorizationAuthenticationIT.kt [hunk 2]
```diff
@@ -96,16 +94,6 @@ class AuthorizationAuthenticationIT {
...
-    fun `test login with valid credentials`() {
```
Comment (ElenaSpb) [inline src/test/kotlin/com/epam/brn/integration/AuthorizationAuthenticationIT.kt:100]: so where is `test login with valid credentials`?

File: src/main/kotlin/com/epam/brn/security/JwtUsernameAndPasswordAuthenticationFilter.kt [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+import javax.servlet.http.HttpServletRequest
+import javax.servlet.http.HttpServletResponse
+import org.springframework.security.authentication.AuthenticationManager
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken
+import org.springframework.security.core.Authentication
+import org.springframework.security.core.AuthenticationException
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter
+import org.springframework.security.web.util.matcher.AntPathRequestMatcher
+
+class JwtUsernameAndPasswordAuthenticationFilter(
```
Comment (ElenaSpb) [inline src/main/kotlin/com/epam/brn/security/JwtUsernameAndPasswordAuthenticationFilter.kt:21]: could you provide tests for this functionallity please

File: src/test/kotlin/com/epam/brn/integration/AuthorizationAuthenticationIT.kt (hunk 2 above)
Comment (goopnigoop) [reply src/test/kotlin/com/epam/brn/integration/AuthorizationAuthenticationIT.kt:100]: this test was created for basic authentication

File: src/main/kotlin/com/epam/brn/security/JwtUsernameAndPasswordAuthenticationFilter.kt (hunk 3 above)
Comment (goopnigoop) [reply src/main/kotlin/com/epam/brn/security/JwtUsernameAndPasswordAuthenticationFilter.kt:21]: ok, if it is necessary, I'll create

File: src/test/kotlin/com/epam/brn/integration/AuthorizationAuthenticationIT.kt (hunk 2 above)
Comment (ElenaSpb) [reply src/test/kotlin/com/epam/brn/integration/AuthorizationAuthenticationIT.kt:100]: so we need tests for new implementation. 
what do you think about it?

**Pull request closed without merging**